quix-frontend/client/src/lib/sql-autocomplete/adapter/sql-autocomplete-adapter-utills.ts

```diff
@@ -0,0 +1,164 @@
+import { findLastDotIndex } from "../../runner/services/autocomplete/highlight-and-score";
+import { TableInfo } from "../sql-context-evaluator";
+import { trinoToJs } from "./trinoToJs"
+
+interface ObjectChild {
+  name: string;
+  dataType: any;
+}
+
+export function getObjectChildren(obj: Record<string, any>, parentName = ''): ObjectChild[] {
+  const children: ObjectChild[] = [];
+
+  for (const key in obj) {
+    const childName = parentName ? `${parentName}.${key}` : key;
+    if (typeof obj[key] === 'object' && obj[key] !== null) {
+      children.push({ name: childName, dataType: obj[key] });
+      children.push(...getObjectChildren(obj[key], childName));
+    } else {
+      children.push({ name: childName, dataType: obj[key] });
+    }
+  }
+  return children;
+}
+
+export function findRelevantPartOfPrefix(tables: TableInfo[], brokenPrefix: string[]): string {
+  // Remove empty cells from brokenPrefix array
+  brokenPrefix = brokenPrefix.filter(cell => cell.trim() !== '');
+
+  const relevantPrefixes: string[] = [];
+
+  for (const table of tables) {
+    for (const column of table.columns) {
+      let found = false;
+      let gotRelevantPartOfPrefix = false;
+      let currentPrefix = '';
+
+      brokenPrefix.forEach(cell => {
+        if (found) {
+          currentPrefix += cell + '.';
+        }
+        if (typeof column === 'object' && !gotRelevantPartOfPrefix && cell === column.name) {
+          found = true;
+          gotRelevantPartOfPrefix = true;
+          currentPrefix += cell + '.';
+        }
+      });
+
+      if (currentPrefix) {
+        relevantPrefixes.push(currentPrefix);
+      }
+    }
+  }
+
+  if (relevantPrefixes.length === 0) {
+    return ''; // No relevant prefixes found
+  }
+
+  // Find the longest relevant prefix
+  let longestPrefix = relevantPrefixes[0];
+  for (const prefix of relevantPrefixes) {
+    if (prefix.length > longestPrefix.length) {
+      longestPrefix = prefix;
+    }
+  }
+
+  return longestPrefix
+}
+
+
+
+function getAllChildrenOfTables(tables: TableInfo[]) {
+  const allChildren = [];
+
+  for (const extractedTable of tables) {
+    const { columns } = extractedTable;
+
+    for (const column of columns) {
+      if (typeof column === 'object') {
```

Comment:
Move this if to filter. columns.filter(c=>c typeof === object)

Reply:
fixed!

Reply:
not yet :)

Reply:
```
function getAllChildrenOfTables(tables: TableInfo[]) {
  const allChildren = [];

  for (const extractedTable of tables) {
    const { columns } = extractedTable;

    (columns.filter(c => typeof c === 'object') as Column[])
      .forEach(({ name, dataType }) => {
        if (typeof dataType === 'string') {
          dataType = trinoToJs(dataType, 0);
        }
        if (typeof dataType === 'object') {
          allChildren.push(...getObjectChildren(dataType, name));
        }
      });
  }

  return allChildren;
}

like this?

Reply:
fine